### PR TITLE
[fix](nereids)support nulls first/last in order by clause

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.88 seconds
 load time: 647 seconds
 storage size: 17123712463 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221230142443_clickbench_pr_71915.html

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -166,7 +166,7 @@ sortClause
     ;
 
 sortItem
-    :  expression ordering = (ASC | DESC)?
+    :  expression ordering = (ASC | DESC)? ( NULLS FIRST | NULLS LAST )?
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:169]: nit:
```suggestion
    :  expression ordering = (ASC | DESC)? (NULLS (LAST | FIRST))?
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -1076,8 +1076,7 @@ public List<String> visitIdentifierSeq(IdentifierSeqContext ctx) {
     public OrderKey visitSortItem(SortItemContext ctx) {
         return ParserUtils.withOrigin(ctx, () -> {
             boolean isAsc = ctx.DESC() == null;
-            // TODO(wj): isNullFirst
-            boolean isNullFirst = true;
+            boolean isNullFirst = !(ctx.NULLS() != null && ctx.LAST() != null);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:1079]: nit:
```suggestion
            boolean isNullFirst = ctx.LAST() == null;
```

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:169]: `(NULLS (FIRST | LAST)` is better

**Review by morrySnow — APPROVED** [review]

**Pull request merged**